### PR TITLE
More fixes for Python 3 compatibility

### PR DESCRIPTION
**Pull request by cclauss**
Please use [PX4 Discuss](http://discuss.px4.io/) or [Slack](http://slack.px4.io/) to align on pull requests if necessary. You can then open draft pull requests to get early feedback.

**Describe problem solved by the proposed pull request**
A clear and concise description of the problem this proposed change will solve. E.g. For this use case I ran into...

**Test data / coverage**
How was it tested? What cases were covered? Logs uploaded to https://review.px4.io/ and screenshots of the important plot parts.

**Describe your preferred solution**
A clear and concise description of what you have implemented.

**Describe possible alternatives**
A clear and concise description of alternative solutions or features you've considered.

**Additional context**
Add any other related context or media.

### CODE REVIEW

**Comment by julianoes** [comment]
Thanks a lot @cclauss. We should get this in right after the release!

File: platforms/nuttx/Debug/Nuttx.py [hunk 1]
```diff
@@ -575,49 +575,49 @@ def invoke(self,args,sth):
...
 			tasks = tasks + self.find_tcb_list(cursor)
-		
+
 		# filter for tasks that are listed twice
 		tasks_filt = {}
 		for t in tasks:
 			pid = parse_int(t['pid']);
 			if not pid in tasks_filt.keys():
-				tasks_filt[pid] = t['name']; 
+				tasks_filt[pid] = t['name'];
```
Comment (cclauss) [inline platforms/nuttx/Debug/Nuttx.py]: Can we lose the semicolon?  This is not JavaScript.

**Comment by julianoes** [comment]
@cclauss I've rebased this, fixed the conflict and force-pushed. Let's see what CI says.

File: Tools/upload_log.py [hunk 2]
```diff
@@ -38,12 +38,7 @@ def ask_value(text, default=None):
     if default != None:
         ask_string += ' (Press ENTER to use ' + default + ')'
     ask_string += ': '
-
-    if sys.version_info[0] < 3:
-        ret = raw_input(ask_string)
-    else:
-        ret = input(ask_string)
-
+    ret = input(ask_string).strip()
```
Comment (julianoes) [inline Tools/upload_log.py]: @cclauss doesn't this require the import magic for `input`?

Comment (cclauss) [reply Tools/upload_log.py]: __from six.moves import input__ is all the _magic_ that is required.

Comment (julianoes) [reply Tools/upload_log.py]: aha, thanks!

**Comment by cclauss** [comment]
There are still three undefined name...

[flake8](http://flake8.pycqa.org) testing of https://github.com/PX4/Firmware on Python 3.8.0

$ __flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics__
```
./Tools/uorb_graph/create.py:369:70: F821 undefined name 'path'
                print('Failed reading file: %s, skipping content.' % path)
                                                                     ^
./platforms/nuttx/Debug/Nuttx.py:419:14: F821 undefined name 'regmap'
		for reg in regmap:
             ^
./platforms/nuttx/Debug/Nuttx.py:420:24: F821 undefined name 'regmap'
			hex_addr= hex(int(a[regmap[reg]]))
                       ^
3     F821 undefined name 'path'
3
```
https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead these tests are focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python a __NameError__ is raised which will halt/crash the script on the user.

**Comment by julianoes** [comment]
Thanks @cclauss, see https://github.com/PX4/Firmware/pull/13763.

**Comment by julianoes** [comment]
It's odd that this now fails in CI after all:
```
[2019-12-19T10:21:36.058Z] + python msg/tools/uorb_to_ros_rtps_ids.py -i msg/tools/uorb_rtps_message_ids.yaml -o px4_ros_com/templates/uorb_rtps_message_ids.yaml
[2019-12-19T10:21:36.327Z] Traceback (most recent call last):
[2019-12-19T10:21:36.327Z]   File "msg/tools/uorb_to_ros_rtps_ids.py", line 11, in <module>
[2019-12-19T10:21:36.327Z]     import six
[2019-12-19T10:21:36.327Z] ImportError: No module named six
script returned exit code 1
```

**Pull request merged**